### PR TITLE
Add diagnostic testing crates

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Replaces #1774.

This PR is entirely about adding two new crates unrelated to any crates we already have. So this PR is about getting generally familiar with it if desired, but also feedback on if we want this to be outside of our normal build and testing, etc.

This embodies the work I did several months ago to create a TUI for diagnosing gossip issues. Creates `holochain_diagnostics` crate which can be our repository for diagnostic testing tooling and patterns. Actual diagnostic tests live in a separate crate `diagnostic_tests`. Devs can create their own diagnostic tests locally using `holochain_diagnostics`, and if they have a good one that seems worth sharing, they can put it in `diagnostic_tests`.

It's expected that diagnostic tests are largely built for the purpose of examining some specific behavior and may not often be reused, but we hope to reuse as much as possible of the patterns or actual tests.

You can run this diagnostic test with `cargo run --release --example link_storm`

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_diagnostics/src/ui/gossip_dashboard/state.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (neonphog) [inline crates/holochain_diagnostics/src/ui/gossip_dashboard/state.rs:1]: empty file?

File: crates/diagnostic_tests/src/zomes.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                })?;
+
+                // let res = api.call(vec![Call {
+                //     target: CallTarget::NetworkAgent(agents[1].clone()),
+                //     zome_name: "zome".into(),
+                //     fn_name: "recv_remote_signal".into(),
+                //     cap_secret: None,
+                //     payload: msg.into(),
+                // }])?;
+                // dbg!(res);
```
Comment (neonphog) [inline crates/diagnostic_tests/src/zomes.rs:118]: There's a lot of commented code throughout this PR

**Review by neonphog — COMMENTED** [review]
@maackle - maybe you could attach a brief video actually running one of the diagnostic tests you wrote?

**Comment by jost-s** [comment]
Yes, or the command to run the diagnostics framework. I've forgotten how it's run.

**Review by jost-s — APPROVED** [review]

**Comment by maackle** [comment]
@neonphog @jost-s good point. You can run it with `cargo run --release --example link_storm`, and a video of an earlier version that I made with Jost is at https://www.youtube.com/watch?v=ciS3USuhdSg

File: crates/diagnostic_tests/src/zomes.rs (hunk 2 above)
Comment (maackle) [reply crates/diagnostic_tests/src/zomes.rs:118]: Sorry I need to get better about cleaning these things up

**Review by neonphog — APPROVED** [review]
Cool. Yeah, if this is helping you I don't see need to audit it overly much.

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**